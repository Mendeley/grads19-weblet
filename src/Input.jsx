import React from "react";

//Use 
//<Input label="name:" type="text" etc /> 
const Input = ({ label, type, id, value, onChange, ...props }) => {
  return (
    <label>
      {label}
      < input type={type} id={id} value={value} onChange={onChange} {...props} />
    </label>
  )
};

export default Input;
