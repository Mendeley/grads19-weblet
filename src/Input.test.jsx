import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import Input from "./Input";

configure({ adapter: new Adapter() });

describe("Input", () => {
  it("renders a text input", () => {
    expect.assertions(1);

    const component = shallow(<Input type="text" />);

    expect(component).toMatchSnapshot();
  });
  it("renders a dateTime input", () => {
    expect.assertions(1);

    const component = shallow(<Input type="datetime-local" />);

    expect(component).toMatchSnapshot();
  });
  it("renders a submit input", () => {
    expect.assertions(1);

    const component = shallow(<Input type="submit" />);

    expect(component).toMatchSnapshot();
  });
});
