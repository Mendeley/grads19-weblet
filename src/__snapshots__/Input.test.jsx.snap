// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders a dateTime input 1`] = `
"<label>
  <input type=\\"datetime-local\\" id={[undefined]} value={[undefined]} onChange={[undefined]} />
</label>"
`;

exports[`Input renders a submit input 1`] = `
"<label>
  <input type=\\"submit\\" id={[undefined]} value={[undefined]} onChange={[undefined]} />
</label>"
`;

exports[`Input renders a text input 1`] = `
"<label>
  <input type=\\"text\\" id={[undefined]} value={[undefined]} onChange={[undefined]} />
</label>"
`;
